ratis-grpc/src/main/java/org/apache/ratis/grpc/server/GrpcLogAppender.java

```diff
@@ -66,6 +68,8 @@
 
   private final GrpcServerMetrics grpcServerMetrics;
 
+  private final ReadWriteLock lock;
```

Comment:
Let's use AutoCloseableReadWriteLock instead of ReentrantReadWriteLock.  Then, we can use try-with-resource to acquire and release the lock.

Reply:
make sense, thanks @szetszwo for the reivew,  i will do it in a new commit